tasks/facial_landmark_detection/search_space.py

```diff
@@ -0,0 +1,359 @@
+                    ['ir_r1_k3_s1_e6_c320']
+                ],
+                'channel_multiplier': 0.75,
+                'depth_multiplier': 0.75
+            },
+            {
+                'arch_def': [
+                    ['ds_r1_k3_s1_c16'],
+                    ['ir_r2_k3_s2_e6_c24'],
+                    ['ir_r3_k3_s2_e6_c32'],
+                    ['ir_r4_k3_s2_e6_c64'],
+                    ['ir_r3_k3_s1_e6_c96'],
+                    ['ir_r3_k3_s2_e6_c160'],
+                    ['ir_r1_k3_s1_e6_c320']
+                ],
+                'channel_multiplier': 0.5,
+                'depth_multiplier': 0.5
+            },
+            {
+                'arch_def': [
+                    ['ds_r1_k3_s1_c16'],
+                    ['ir_r2_k3_s2_e6_c24'],
+                    ['ir_r3_k3_s2_e6_c32'],
+                    ['ir_r4_k3_s2_e6_c64'],
+                    ['ir_r3_k3_s1_e6_c96'],
+                    ['ir_r3_k3_s2_e6_c160'],
+                    ['ir_r1_k3_s1_e6_c320']
+                ],
+                'channel_multiplier': 1.25,
+                'depth_multiplier': 1.25
+            }
+        ]
+
+        self.config_all = {}
+        self.arch_counter= 0
+        self.num_classes = num_classes
+        self.r_range = tuple(args.r_range)
+        self.e_range = tuple(args.e_range)
+        self.k_range = tuple(args.k_range)
+        self.channel_mult_range = tuple(args.channel_mult_range)
+        self.depth_mult_range = tuple(args.depth_mult_range)
+        #make each run deterministic
+        torch.manual_seed(0)
+        random.seed(0)
+        np.random.seed(0)
+
+    @overrides
+    def random_sample(self)->ArchaiModel:
+        ''' Uniform random sample an architecture, always start with a few seed models'''
+
+        if (self.arch_counter >= 0 and self.arch_counter <= 3):
+            cfg = copy.deepcopy(self.cfgs_orig[self.arch_counter])
+            arch = self._create_uniq_arch(cfg)
+        else:
+            arch = None
+            while (arch == None) : 
+                cfg = self._rand_modify_config(self.cfgs_orig[0], len(self.e_range), len(self.r_range), len(self.k_range), 
+                                                    len(self.channel_mult_range), len(self.depth_mult_range))
+                arch = self._create_uniq_arch(cfg)
+            assert (arch != None)
+
+        logger.info(f"{sys._getframe(0).f_code.co_name} return archid = {arch.archid} with config = {arch.metadata}")
+        
+        return ArchaiModel(arch=arch, archid=arch.archid, metadata={'config': arch.metadata})
+
+    @overrides
+    def save_arch(self, model: ArchaiModel, file: str):
+        with open(file, 'w') as fp:
+            cfg = model.metadata['config']
+            json.dump({'config': cfg}, fp)
+
+    @overrides
+    def load_arch(self, file: str):
+        metadata = json.load(open(file))
+        config = json.loads(metadata['config'])
+        arch = ConfigSearchModel(config)
```

Comment:
I don't understand this construction of ConfigSearchModel - the __init__ function on ConfigSearchModel  takes 3 args, `model : ArchaiModel, archid: str, metadata : dict` ?  And these 3 args I don't get since they mirror ArchaiModel itself. I think everywhere you use ConfigSearchModel  you could just use ArchaiModel.  I also wonder how much of this could be simplified by using `ArchConfig`

Reply:
Wow. I can't believe I missed this. The ConfigSearchModel was a class I had before ArchaiModel was avaiable. It is redundant now. I will remove it and please do take another look to double check.

Reply:
Fixed.